plugins/parsers/prometheus/parser.go

```diff
@@ -28,12 +33,28 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	// Read raw data
 	buffer := bytes.NewBuffer(buf)
 	reader := bufio.NewReader(buffer)
+	mediatype, params, err := mime.ParseMediaType(p.Header.Get("Content-Type"))
```

Comment:
This will fail if `p.Header` is not set. Please handle this e.g. by
```suggestion
	mediatype, params, err := "", nil, nil
	if p.Header != nil {
		mediatype, params, err = mime.ParseMediaType(p.Header.Get("Content-Type"))
	}
```